### PR TITLE
Convert WebServerHelper.java to Kotlin

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes: #2986
Parent Issue: #1826 

Converted WebServerHelper.java to Kotlin

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      .filter { s: String? -> s != INVALID_IP }
+      .timeout(15, TimeUnit.SECONDS)
+      .take(1)
+      .observeOn(AndroidSchedulers.mainThread())
+      .subscribe(
+        { s: String? ->
+          ipAddressCallbacks.onIpAddressValid()
+          Log.d(TAG, "onSuccess:  $s")
+        }
+      ) { e: Throwable? ->
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt:94]: We can get rid of the redundant lambda arrows.

**Review by gouri-panda — APPROVED** [review]

**Comment by kelson42** [comment]
@gouri-panda I don’t understand, you approve but request changes!

**Comment by gouri-panda** [comment]
@kelson42 These are neat changes :) We can merge if we want to.

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * Created by Adeel Zafar on 18/07/2019.
+ */
+class WebServerHelper @Inject constructor(
+  private val kiwixServerFactory: KiwixServer.Factory,
+  private val ipAddressCallbacks: IpAddressCallbacks
+) {
+  private var kiwixServer: KiwixServer? = null
+  private var isServerStarted = false
+
+  @Suppress("ReturnCount")
```
[the remark this answers is not in the document]
Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt]: hi @gouri-panda done

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    if (isServerStarted) {
+      kiwixServer?.stopServer()
+      updateServerState(false)
+    }
+  }
+
+  @Suppress("MagicNumber")
+  private fun startAndroidWebServer(selectedBooksPath: ArrayList<String>): Boolean {
+    if (!isServerStarted) {
+      ServerUtils.port = 8080
```
[the remark this answers is not in the document]
Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt]: Done

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt (hunk 1 above)
Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt]: Done

**Comment by kelson42** [comment]
@MohitMaliDeveloper @gouri-panda Codecov does not pass... and now it should pass all the time. Where is the problem (in the new codecov conf or in the PR)?

**Comment by gouri-panda** [comment]
@kelson42 Here the patch's coverage is 12.82% but we are targeting 100% in the codecov config also the same with the project's default threshold.

**Comment by kelson42** [comment]
We probably need to make an exception

**Pull request merged**